Beaglebone: support weston with fbdev

Ponsuke07 opened the pull request:
Hi EMLinux team,

I created patches to support weston with fbdev on beaglebone.

## How to build

Add MACHINE ?= "beaglebone" to local.conf, and

```
bitbake core-image-weston
```

## Test

I confirmed that weston was worked on Beaglebone black and black-wireless.
 
Thank you

Ponsuke07 committed:
weston-conf: Use fbdev-backend for BeagleBone Black

Ponsuke07 committed:
kernel: beaglebone: Enable NXP Semiconductors TDA998X HDMI encoder

BeagleBone Black has TDA998X HDMI encoder. Let's enable it to support
HDMI output.

Ponsuke07 committed:
machine: beaglebone: Add kernel-modules to core-image-weston

This patch add kernel-modules to MACHINE_EXTRA_RRECOMMENDS so that
can be probed modules which are required by weston.

Note that kernel-modules won't be installed in core-image-minimal.

hiraku-wfs commented on recipes-graphics/weston/weston-conf.bbappend:
```diff
@@ -20,4 +20,12 @@ do_install_raspberrypi3-64() {
+[core]
+backend=fbdev-backend.so
+EOF
+}
```
Thanks a lot!
If you have time, could you make a common function for the use of fbdev-backend? 
(If not, I will just merge current this PR.)

```
do_install_fbdev_backend () {
        mkdir -p ${D}/${sysconfdir}/xdg/weston
        cat << EOF > ${D}/${sysconfdir}/xdg/weston/weston.ini
[core]
backend=fbdev-backend.so
EOF
}

do_install_qemuarm () {
        do_install_fbdev_backend
}
...
do_install_beaglebone () {
        do_install_fbdev_backend
}
```

Ponsuke07 replied:
Yes! I think it's not a good implementation, too. 
So I rewrite weston-conf.bbappend, please look 0a4a6a0.

Ponsuke07 replied:
> do_install_fbdev_backend () {
...(snip)...
> do_install_beaglebone () {
>         do_install_fbdev_backend
> }

I'm sorry to ignore your sample code. I just realized it. 
Thank you for your suggestions. 

However, I feel the your code is a bit messy because it needs to add or fix (when user want to use backend other than fbdev) do_install() for each board.
(I know it's same way of thinking of poky-warrior)
my patch override do_install() in poky layer, however it has compatible.

hiraku-wfs replied:
Your code is better than mine. Thank you.
I think that default value of the DEFAULTBACKEND should be "fbdev" instead of empty string so that we can skip the following check.

```
    if [ -n "${DEFAULTBACKEND}" ]; then
```

Ponsuke07 replied:
>  I think that default value of the DEFAULTBACKEND should be "fbdev" instead of empty string so that we can skip the following check.

I think do_install() should be skipped by default (in other words DEAFULTBACKEND should be set empty by default)
because do_install() in poky's weston-conf.bb was written to do nothing by default[1] other than qemux86/qemux86-64.
Otherwise my patch may have a negative effect when built on other machines.

[1] https://git.yoctoproject.org/cgit/cgit.cgi/poky/tree/meta/recipes-graphics/wayland/weston-conf.bb?h=warrior

hiraku-wfs replied:
Got it. I will merge this PR.
Thank you!

Ponsuke07 committed:
weston-conf: Make do_install() to be common

This patch make do_install() to be common.
And Introduce DEFAULTBACKEND variable, so that user can set appropriate
backend.

All qemu machines, raspberrypi3-64 and beaglebone are set to fbdev by
default.

Ponsuke07 commented:
Thank you for accepting my PR!
